backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/user/UserServiceApplicational.java

```diff
@@ -23,33 +26,38 @@
     @Autowired
     private UserRepository userRepository;
 
+    @Autowired
+    private AuthUserRepository authUserRepository;
+
     @Value("${spring.mail.username}")
     private String mailUsername;
 
     public ExternalUserDto createExternalUser(Integer courseExecutionId, ExternalUserDto externalUserDto) {
-        boolean userExists = userRepository.findByUsername(externalUserDto.getEmail()).isPresent();
+        boolean userExists = authUserRepository.findAuthUserByUsername(externalUserDto.getEmail()).isPresent();
         ExternalUserDto user = userService.createExternalUserTransactional(courseExecutionId, externalUserDto);
         if (!userExists) {
-            sendConfirmationEmailTo(user);
+            sendConfirmationEmailTo(user.getEmail(), user.getConfirmationToken());
         }
         return user;
     }
 
     public NotificationResponse<CourseDto> importListOfUsers(InputStream stream, int courseExecutionId) {
         NotificationResponse<CourseDto> courseDtoNotificationResponse = userService.importListOfUsersTransactional(stream, courseExecutionId);
         courseDtoNotificationResponse.getResponse().getCourseExecutionUsers()
-                .forEach(this::sendConfirmationEmailTo);
+                .stream()
+                .map(UserDto::getUsername)
+                .map(authUserRepository::findAuthUserByUsername)
```

Comment:
Why doing this? It is possible to trust userService.importListOfUsersTransactional